packages/bitcore-client/src/wallet.ts

```diff
@@ -360,16 +360,21 @@ export class Wallet {
       }
       tokenContractAddress = tokenObj.address;
     }
+    let change = params.change;
+    if (change === 'miner') {
+      change = undefined; // no change
+    } else if (!change) {
+      change = await this._getChangeAddress();
```

Comment:
This change causes the wallet to need to be unlocked before the newTx method is called. Otherwise it gives an exception. In the case of wallet-send you just need to move the unlock statement up above the call to newTx. Looks like wallet-bump-tx, wallet-payrpo and wallet-tx also use this method.

Reply:
The reason why is because the call to derivePrivateKey uses `this.unlocked.masterKey` and `this.unlocked` is undefined before unlocking the wallet.